app/controllers/queue_items_controller.rb

```diff
@@ -33,4 +42,14 @@ def new_queue_item_position
   def video_already_in_my_queue?(video)
     current_user.queue_items.map(&:video).include?(video)
   end
+  
+  def update_queue_items
+    ActiveRecord::Base.transaction do
+      params[:queue_items].each do |queue_item_data|
+        queue_item = QueueItem.find(queue_item_data[:id])
+        queue_item.update_attributes!(position: queue_item_data[:position], rating: queue_item_data[:rating]) if queue_item.user == current_user
```

Comment:
Personally I would probably move this to a standard `if` statement as the line is getting a bit long.